### DISCUSSION TITLE
Equivalent of `coroutine.yield(value)` from rust ?

### DISCUSSION THREAD

**Discussion by tguichaoua** [mlua-rs/mlua#164] [Q&A] [answered]
How to call / what is the equivalent of `coroutine.yield(value)` from Rust code ?
I also want to pass an UserData to the yield.

**Comment by khvzak** [comment]
Unfortunately you cannot yield from Rust code as Lua does not allow to yield across C-call boundary (which is applicable to Rust).
The only way to yield is to use Rust async functions and [`tokio::task::yield_now`](https://docs.rs/tokio/latest/tokio/task/fn.yield_now.html)

**Comment by tguichaoua** [comment, marked as answer]
Since it's not possible to yield accros C-call boundary, I move my api into lua.
```lua
function foo()
    local ud = _api.foo() -- call rust to create the userdata
    coroutine.yield(ud)
end
```
